vpr/src/route/rr_graph_fpga_interchange.cpp

```diff
@@ -0,0 +1,1706 @@
+                resistance = get_corner_value(model.getResistance(), "slow", "typ");
+            }
+            node_to_RC_[node_id].first += resistance;
+            node_to_RC_[node_id].second += capacitance;
+#ifdef DEBUG
+            if (constant_node) {
+                const auto& wires = ar_.getWires();
+                std::tuple<int, int> base_wire_(wires[node.getWires()[0]].getTile(), wires[node.getWires()[0]].getWire());
+                VTR_LOG("Constant_node: %s\n", wire_to_node_[base_wire_] == 1 ? "VCC\0" : "GND\0");
+                for (const auto& wire_id_ : node.getWires()) {
+                    const auto& wire = ar_.getWires()[wire_id_];
+                    int tile_id = wire.getTile();
+                    int wire_id = wire.getWire();
+                    VTR_LOG("tile:%s wire:%s\n", str(tile_id).c_str(), str(wire_id).c_str());
+                }
+            }
+#endif
+            for (const auto& wire_id_ : node.getWires()) {
+                const auto& wire = ar_.getWires()[wire_id_];
+                int tile_id = wire.getTile();
+                int wire_id = wire.getWire();
+                wire_to_node_[std::make_tuple(tile_id, wire_id)] = node_id;
+                node_to_locs_[node_id].insert(tile_to_loc_[tile_id]);
+                if (wire_name_to_seg_idx_.find(str(wire_id)) == wire_name_to_seg_idx_.end())
+                    wire_name_to_seg_idx_[str(wire_id)] = -1;
+                node_tile_to_segment_[std::make_tuple(node_id, tile_id)] = wire_name_to_seg_idx_[str(wire_id)];
+            }
+        }
+        total_node_count_ = id;
+    }
+
+    /*
+     * Create entry for each pip in device architecture,
+     * also store meta data related to that pip such as: tile_name, wire names, and its direction.
+     */
+    void create_pips_() {
+        for (const auto& tile : ar_.getTileList()) {
+            int tile_id = tile.getName();
+            location loc = tile_to_loc_[tile_id];
+            const auto& tile_type = ar_.getTileTypeList()[tile.getType()];
+
+            for (const auto& pip : tile_type.getPips()) {
+                int wire0_name, wire1_name;
+                int node0, node1;
+                int switch_id;
+
+                wire0_name = tile_type.getWires()[pip.getWire0()];
+                wire1_name = tile_type.getWires()[pip.getWire1()];
+                if (wire_to_node_.find(std::make_tuple(tile_id, wire0_name)) == wire_to_node_.end())
+                    continue;
+                if (wire_to_node_.find(std::make_tuple(tile_id, wire1_name)) == wire_to_node_.end())
+                    continue;
+                node0 = wire_to_node_[std::make_tuple(tile_id, wire0_name)];
+                node1 = wire_to_node_[std::make_tuple(tile_id, wire1_name)];
+                // Allow for pseudopips that connect from/to VCC/GND
+                if (pip.isPseudoCells() && (node0 > 1 && node1 > 1))
+                    continue;
```

Comment:
I think we can allow pseudo PIPs by default

Reply:
We could but this will create incorrect designs.
For now we don't create physical netlist so this shouldn't be an issue

Reply:
The main issue is that disallowing pseudo PIPs in general will make some devices completely unroutable (such as the IOB33 to General Interconnect problem for xc7)